### ISSUE TITLE
Change API Key in realtime

### ISSUE THREAD

**Issue by zmsansan** [algolia/instantsearch-android#63]
Hello! I need change API Key in realtime, but it impossible.
This need if user be very intensive use search or if key leak, i'am banned this user, and need change secure key for all users in realtime.

**Comment by PLNech** [comment]
HI @zmsansan, thanks for your issue report!

Indeed, changing API keys in realtime is not something we support for now. I'm adding it to our backlog, I'll update you on this page as we move forward on this feature!

In the meantime, you can simply reinstantiate the `InstantSearch` instance with a new `Searcher` that uses the new API Key:

```java
// Init with the first API KEY
searcher = Searcher.create(ALGOLIA_APP_ID, ALGOLIA_API_KEY, ALGOLIA_INDEX_NAME);
instantSearch = new InstantSearch(this, searcher); 

// Reset with a new API KEY (this will connect the widgets to the new Searcher)
searcher = Searcher.create(ALGOLIA_APP_ID, ALGOLIA_OTHER_API_KEY, ALGOLIA_INDEX_NAME);
instantSearch = new InstantSearch(this, searcher); 
```
From this point, the SearchBox will send queries to the new searcher which will dispatch it to the other widgets.

**Comment by zmsansan** [comment]
@PLNech i'am use version 1.4.0 (1.4.1 dont installed from repository) and Searcher is Singleton. But i'am copy Searcher class and did no Singleton (and change dependent classes) and did how you say, its work, thanks!

**Comment by PLNech** [comment]
@zmsansan: Ah indeed, we are still reviewing the refactoring of `Searcher` to make it instantiable several times. Glad the workaround worked. I'll keep this issue opened and will update you here as soon as the refactoring is released!

**Comment by PLNech** [comment]
Hi @zmsansan , we just released [v1.5.0](https://github.com/algolia/instantsearch-android/releases/tag/1.5.0) which ships the ability to instantiate several `Searchers`.

Updating to this version should be enough to avoid needing your Searcher copy anymore.

Should you have any problem with using this, please reopen this issue to let me know and I'll investigate your situation 🙂
